lib/logger/lib/logger.ex

```diff
@@ -573,7 +573,7 @@ defmodule Logger do
     if compare_levels(level, min_level) != :lt do
       macro_log(level, data, metadata, caller)
     else
-      :ok
+      handle_unused_variable_warnings(data)
```

Comment:
It depends on what we want to do with the compile time purge. Do we want to always execute the expression on the right hand side? Wouldn't that partially defeat the purpose of compile time purging?